lib/excon.rb

```diff
@@ -42,13 +42,23 @@ class << self
 
     # @return [Hash] defaults for Excon connections
     def defaults
-      @defaults ||= DEFAULTS
+      defaults = Thread.current.thread_variable_get(:excon_defaults)
```

Comment:
wonder if the variable name should be more complex, since the namespace is (thread-)global. ideally ruby would have gensyms, but nope.